src/ory-editor/converter/src/createPlugin.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Created by benny on 17.11.16.
+ */
+import { v4 } from 'uuid'
+import markdownToSlate from './markdownToSlate'
+
+const createPlugins = ({ normalized, elements }) => {
+  const split = normalized
+    .split(/(§\d+§)/)
+    .map(s => s.trim())
+    .filter(s => s !== '')
+  return split.map(markdown => {
+    // console.log(markdown)
+    var elementID = /§(\d+)§/.exec(markdown)
+    // console.log(elementID)
+    if (elementID !== null) {
+      return {
+        cells: [createPluginCell(elements[elementID[1]])]
+      }
+    } else {
+      return {
+        cells: [
+          {
+            markdown: markdown
+          }
+        ]
+      }
+    }
+  })
+}
+const createPluginCell = elem => {
+  switch (elem.name) {
+    case 'table':
+      return {
+        content: {
+          plugin: {
+            name: 'serlo/content/markdown'
+          },
+          state: {
+            src: elem.src
+          }
+        }
+      }
+    case 'spoiler':
+      const rows = createPlugins(elem.content)
+
+      return {
+        content: {
+          plugin: {
+            name: 'ory/editor/core/layout/spoiler'
+          },
+          state: {
+            title: elem.title,
+            content: {
+              type: '@splish-me/editor-core/editable',
+              // FIXME: can we move that before createPlugin?
+              state: markdownToSlate({
```

Comment:
@Knorrke: could we do `markdownToSlate` before `createPlugins`? You see that I need it to create the state for the new "layout" plugins. And since the state structure of "layout" plugins could be whatever, the later called `markdownToSlate` doesn't recognise the cases. My workaround works but seems kinda hacky